### PR TITLE
Added new module csvtk/join

### LINKED ISSUE #3705: new module: csvtk/join
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by anoronh4**
Joined new module csvtk/join

## PR checklist

Closes #3705 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/csvtk/join/main.nf
Comment (nvnieuwk) [inline modules/nf-core/csvtk/join/main.nf:13]: These are not mandatory arguments of the tool, see further for suggestions on how to implement this in another way

File: modules/nf-core/csvtk/join/main.nf
Comment (nvnieuwk) [inline modules/nf-core/csvtk/join/main.nf:25]:
```suggestion
    def delimiter = csv.extension == "tsv" ? "\t" : ","
```
This gets the extension of the input file and defines the delimiter

File: modules/nf-core/csvtk/join/main.nf
Comment (nvnieuwk) [inline modules/nf-core/csvtk/join/main.nf:27]:
```suggestion
    out_extension = args.contains('--out-delimiter "\t"') || args.contains('-D "\t"' ? "tsv" : "csv"
```
This looks if `--out-delimiter` or `-D` is in the args. If `\t` is the option it will use `tsv` as the output extension, otherwise it will use `csv` (which is the default delimiter)

File: modules/nf-core/csvtk/join/main.nf
Comment (nvnieuwk) [inline modules/nf-core/csvtk/join/main.nf:34]:
```suggestion
```

File: modules/nf-core/csvtk/join/main.nf
Comment (nvnieuwk) [inline modules/nf-core/csvtk/join/main.nf:48]: You should also add the output extension check here (stub runs will not run the `script` block so the extension will be undefined here)

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

**Review by luisas — APPROVED** [review]

File: modules/nf-core/csvtk/join/main.nf
Comment (anoronh4) [reply modules/nf-core/csvtk/join/main.nf:25]: @nvnieuwk i didn't have a reason behind the logic initially, i was just following the convention of other csvtk modules for consistency. but now that i'm reviewing it, doesn't it make sense to have users configure the delimiter in ext.args? people often use alternative delimiters, or use the `txt` extension when their file is tab-delimited.

Comment (anoronh4) [reply modules/nf-core/csvtk/join/main.nf:25]: in some ways i like the old format because it makes the in and out delimiters more deliberate coming from the user, and the module doesn't have to guess which delimiter will be used based on string matching.

**Comment by anoronh4** [comment]
i'm not sure why merging is still blocked. looks like i marked each change request was addressed. possibly still needs approval from the requester.

**Review by luisas — APPROVED** [review]

**Review by nvnieuwk — APPROVED** [review]
I'm sorry I was on holiday :), you can always ping someone from the maintainers team in the nf-core Slack if you get no response

**Pull request merged**
